### ISSUE TITLE
Read-only-mount guard is Cowork-only, so the "never edit skill files in place" rule has no enforcement in Claude Code

### ISSUE THREAD

**Issue by seqorasecurity** [rebelytics/one-skill-to-rule-them-all#23]
## Summary

The skill tells the agent never to edit a live skill file in place, and backs that up by describing the live file as read-only. In Cowork that's true — the mount rejects writes with `EROFS`. In Claude Code the same files are ordinary user-owned files with write permission, and nothing stops the write.

So the guard the skill describes is real in one environment and absent in the other, while the text reads as though it applies everywhere. The result is that confidence in the backstop is highest exactly where the backstop is missing.

This is adjacent to #17 (Cowork-specific assumptions leaking into otherwise environment-agnostic instructions), but it's a different failure: #17 is about tool *names*, this is about an *enforcement guarantee* the agent relies on when deciding how careful to be.

## How it showed up

During a scheduled autonomous run of the Comprehensive Review, the agent opened a live `SKILL.md` and applied an observation to it directly, contrary to the skill's own rule. It was caught and reverted in the same session and no live skill was left modified — but the thing that caught it was the agent noticing, not anything structural.

Two causes, and I think the second is the more interesting one:

**1. The asserted guard doesn't exist in Claude Code.** From "Skill file locations — read-only mount vs workspace copy":

> **The live file is read-only in Cowork.** In Cowork, the live skill file is mounted read-only at `.claude/skills/{skill}/SKILL.md`. You can read it, but you cannot edit it directly — the file system will reject write attempts with `EROFS` (Read-Only File System).

The paragraph is correctly scoped in its first clause and then generalises in the reader's head. Nothing in it says "and in other environments the write will simply succeed."

**2. Step 5's ordering puts the forbidden path in hand.** Step 5 says "create an updated version of its SKILL.md" and names the destination only in an `**Important:**` note at the end of the step. Meanwhile the natural motion of the work — read the live file, then edit what you just read — leaves the live path as the most available thing to act on. The rule is stated, correctly and more than once, and still loses to the ordering.

## Suggested fix

Two changes, both small.

**Scope the read-only claim.** Say which environments have the guard, and state plainly that in a local-filesystem environment the discipline is the only thing preventing the overwrite. Something like: *assume you are in the unprotected case unless you have seen an `EROFS` yourself.*

**Make Step 5 begin with the copy.** Turn the prohibition into the mechanics rather than leaving it as a nearby instruction:

```bash
mkdir -p "[workspace]/skill-updates/[today]/[skill-name]"
cp "<live>/SKILL.md" "[workspace]/skill-updates/[today]/[skill-name]/SKILL.md"
diff -q "<live>/SKILL.md" "[workspace]/skill-updates/[today]/[skill-name]/SKILL.md"
# then make EVERY edit against the staged path
```

This does three jobs at once: the live path is never the target of an edit, the staged copy provably starts from live (which satisfies the existing diff-before-overwrite rule in the same move), and a stale staged copy from an earlier date can't be picked up by accident. Violating the rule then costs an extra deliberate step instead of one fewer.

## Why I think it generalises

This is the skill's own Pre-Flight Principle turned on itself — *does this skill have rules, and does it have a mechanism to enforce them?* For this rule, in this environment, the answer was no.

The broader pattern, which may be worth a line somewhere in the skill: **a rule whose compliance is guaranteed by the environment in one context and by the reader's memory in another will be broken in the second context**, and describing the environmental guard as universal is what makes it likely. And more generally, a prohibition placed *near* a workflow step tends to lose to that step's natural motion; reordering the mechanics so the prohibited path is never the one in hand is the more reliable fix than restating the rule more forcefully.

One smaller data point from the same session: rewriting a file wholesale instead of copy-then-edit also introduced a silent one-word regression, caught only by diffing against the live file. That's a second argument for making `cp` + `diff` the opening move rather than a closing check.

---

Happy to open a PR with both changes if that's useful. Thanks for the skill — the observation-log workflow has been genuinely valuable.

**Comment by rebelytics** [comment]
Thank you, @seqorasecurity. No need to open a PR. I’ll take care of it for v3 and make you co-author.

**Comment by rebelytics** [comment]
Fixed in `release/v3.0.0`. The read-only mount is now named as a Cowork-only guard, and the text tells the agent to assume it's in the unprotected case unless it has actually seen an EROFS. Review Step 5 begins with your sequence — `mkdir -p`, `cp` the live file into the staging dir, `diff -q` to prove it matches — so the live path is never the one in hand when the edit happens.

You're credited as co-author on the commit. FYI: I'll be running v3 for a week or so before it goes into main.

**Comment by rebelytics** [comment]
Shipped in v3.0.0, released today. Thanks again for your contribution — the release notes credit everyone whose reports and PRs shaped this version.
